compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -292,6 +294,8 @@ public void visit(BLangPackage pkgNode, AnalyzerData data) {
         }
         SymbolEnv pkgEnv = this.symTable.pkgEnvMap.get(pkgNode.symbol);
         data.env = pkgEnv;
+        int originalTopLevelNodeSize = pkgNode.topLevelNodes.size();
```

Comment:
Move this closer to where it is used.

Reply:
We need to get the original number of top level nodes. In some instances nodes were added after evaluating constants so that is why this was put before that.

Reply:
Why can't we let the analysis happen for those nodes?

Reply:
Those nodes need to be analysed after analyzing  the original number of nodes to properly resolve. Like in some cases where additionally created nodes were being used inside functions, I got errors when I tried to analyse them before analyzing that function node.

Reply:
The new ones get added at the end, right? Any idea why the newer ones were analyzed before the existing ones?

Reply:
New ones are not exactly getting resolved before the existing ones, but since we are analyzing `NodeKind.TYPE_DEFINITION` after constants before analyzing other nodes in occurrence order, then if there were new nodes that were added at the end that are of type `NodeKind.TYPE_DEFINITION` then without `originalTopLevelNodeSize` those will also get analysed in that for loop. The issue in that is the nodes that get added at the end only gets resolved properly if we first resolve the functions and other nodes that come before them. So I'm using `originalTopLevelNodeSize` to only analyse the original number of `NodeKind.TYPE_DEFINITION` nodes that were present. After the original number of `NodeKind.TYPE_DEFINITION`  are analysed the rest of the nodes will  get analysed as usual in their occurrence order.